test/CMakeLists.txt

```diff
@@ -14,11 +14,15 @@ list(APPEND TEST_SOURCES
 			 test_dbc.cpp
 			 test_utils.cpp)
 
-add_executable(tests ${TEST_SOURCES})
-target_compile_definitions(tests PRIVATE TESTDBCFILES_PATH="${CMAKE_CURRENT_SOURCE_DIR}/dbcs")
-target_link_libraries(tests PRIVATE dbc Catch2::Catch2WithMain)
-target_sources(tests INTERFACE FILE_SET HEADERS
-                               TYPE HEADERS
-                               BASE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}
-                               FILES defines.hpp)
-catch_discover_tests(tests)
+add_executable(dbcParserTests ${TEST_SOURCES})
+target_compile_definitions(dbcParserTests PRIVATE TESTDBCFILES_PATH="${CMAKE_CURRENT_SOURCE_DIR}/dbcs")
+target_link_libraries(dbcParserTests PRIVATE dbc Catch2::Catch2WithMain)
+if (${CMAKE_MINOR_VERSION} GREATER_EQUAL 23)
+    target_sources(dbcParserTests PRIVATE FILE_SET HEADERS
+                                   TYPE HEADERS
+                                   BASE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}
+                                   FILES defines.hpp)
+else()
+    target_include_directories(dbcParserTests PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+endif()

```

Comment:
why is there need for this if statement?

Reply:
The first part is the prefered way to do it, the second is the old approach for cmake less tham 3.23. Problem is that labplot needs a low version